js/repeater.js

```diff
@@ -691,90 +726,71 @@
 			}
 		},
 
-		runRenderer: function (viewTypeObj, data, callback) {
-			var $container, i, l, response, repeat, subset;
-
-			function addItem ($parent, resp) {
-				var action;
-				if (resp) {
-					action = (resp.action) ? resp.action : 'append';
-					if (action !== 'none' && resp.item !== undefined) {
-						$parent = (resp.container !== undefined) ? $(resp.container) : $parent;
-						$parent[action](resp.item);
-					}
-				}
-			}
-
+		renderItems: function renderItems (viewTypeObj, data, callback) {
 			if (!viewTypeObj.render) {
 				if (viewTypeObj.before) {
-					response = viewTypeObj.before.call(this, {
+					var addBefore = viewTypeObj.before.call(this, {
 						container: this.$canvas,
 						data: data
 					});
-					addItem(this.$canvas, response);
+					addItem(this.$canvas, addBefore);
 				}
 
-				$container = this.$canvas.find('[data-container="true"]:last');
+				var $container = this.$canvas.find('[data-container="true"]:last');
 				$container = ($container.length > 0) ? $container : this.$canvas;
 
 				if (viewTypeObj.renderItem) {
-					repeat = viewTypeObj.repeat || 'data.items';
+					var subset;
+					var repeat = viewTypeObj.repeat || 'data.items';
 					repeat = repeat.split('.');
 					if (repeat[0] === 'data' || repeat[0] === 'this') {
 						subset = (repeat[0] === 'this') ? this : data;
 						repeat.shift();
 					} else {
 						repeat = [];
 						subset = [];
-						if (window.console && window.console.warn) {
-							window.console.warn('WARNING: Repeater plugin "repeat" value must start with either "data" or "this"');
-						}
+						logWarn('WARNING: Repeater plugin "repeat" value must start with either "data" or "this"');
 					}
 
-					for (i = 0, l = repeat.length; i < l; i++) {
-						if (subset[repeat[i]] !== undefined){
-							subset = subset[repeat[i]];
+					for (var repeatI = 0, repeatL = repeat.length; repeatI < repeatL; repeatI++) {
+						if (subset[repeat[repeatI]] !== undefined) {
+							subset = subset[repeat[repeatI]];
 						} else {
 							subset = [];
-							if (window.console && window.console.warn) {
-								window.console.warn('WARNING: Repeater unable to find property to iterate renderItem on.');
-							}
-							break;
+							logWarn('WARNING: Repeater unable to find property to iterate renderItem on.');
+							repeatI = repeat.length;
 						}
 					}
 
-					for (i = 0, l = subset.length; i < l; i++) {
-						response = viewTypeObj.renderItem.call(this, {
+					for (var subItemI = 0, subsetL = subset.length; subItemI < subsetL; subItemI++) {
```

Comment:
Man oh man, I went back and forth on restructuring this for clarity/readability as you suggest. Since jsperf was down and because of all the peculiarities of different browsers, and because of how freaking slow the repeater already is, I went with the "safe" (circa 2011) practice of manually caching.

However, re-examining this code this morning, a few things stand out to me:
1) This entire method is only called a _single time_
2) These `for` loops each only iterate a few times (the first one usually just _one time_ and the second only as many rows or thumbnails as are loaded on page, typically less than 100)

Not to mention the link you pointed to calls the entire practice into question (at least for Chrome).

Changing.